pkg/util/provider/machinecontroller/machineclass.go

```diff
@@ -201,16 +171,16 @@ func (c *controller) reconcileClusterMachineClass(ctx context.Context, class *v1
 func (c *controller) addMachineClassFinalizers(ctx context.Context, class *v1alpha1.MachineClass) error {
 	finalizers := sets.NewString(class.Finalizers...)
 	finalizers.Insert(MCMFinalizerName)
-	return c.updateMachineClassFinalizers(ctx, class, finalizers.List())
+	return c.updateMachineClassFinalizers(ctx, class, finalizers.List(), true)
 }
 
 func (c *controller) deleteMachineClassFinalizers(ctx context.Context, class *v1alpha1.MachineClass) error {
```

Comment:
I want to rename this to `deleteMCMFinalizerFromMachineClass`. We can do this later as well, if not with this PR. Similarly `addMCMFinalizerToMachineClass` instead of `addMachineClassFinalizers`

Reply:
done